src/topology/category/Top/limits.lean

```diff
@@ -125,6 +126,393 @@ instance forget_preserves_colimits : preserves_colimits (forget : Top.{u} ⥤ Ty
     by exactI preserves_colimit_of_preserves_colimit_cocone
       (colimit_cocone_is_colimit F) (types.colimit_cocone_is_colimit (F ⋙ forget)) } }
 
+lemma limit_induced (F : J ⥤ Top.{u}) :
+  (limit F).str = ⨅ j, (F.obj j).str.induced (limit.π F j) :=
+begin
+  let homeo := homeo_of_iso (limit.iso_limit_cone ⟨_, limit_cone_is_limit F⟩),
+  refine homeo.inducing.induced.trans _,
+  dsimp only [is_open, limit_cone, of, Top.topological_space, subtype.topological_space,
+    Pi.topological_space],
+  simp only [induced_infi, induced_compose],
+  congr
+end
+
+section prod
+
+lemma prod_topology {X Y : Top} :
+  (X ⨯ Y).topological_space =
+    induced (limits.prod.fst : X ⨯ Y ⟶ _) X.topological_space ⊓
+      induced (limits.prod.snd : X ⨯ Y ⟶ _) Y.topological_space :=
+begin
+  change (X ⨯ Y).str = _,
+  conv_lhs { rw Top.limit_induced },
+  apply le_antisymm,
+  rw le_inf_iff,
+  exact ⟨infi_le _ walking_pair.left, infi_le _ walking_pair.right⟩,
+  rw le_infi_iff,
+  rintro (_|_),
+  exacts [inf_le_left, inf_le_right]
+end
+
+/--
+The isomorphsim between the underlying set of `X ⨯ Y` and the set-theoretic product of `X` and `Y`.
+-/
+def prod_iso_prod (X Y : Top.{u}) : ((X ⨯ Y : Top) : Type*) ≅ X × Y :=
```

Comment:
I think an isomorphism of the form
```lean
def prod_iso_prod (X Y : Top.{u}) : X ⨯ Y ≅ Top.of (X × Y) := sorry
```
would be more useful. The first `X ⨯ Y` is the categorical product in `Top`, while the `Top.of (X × Y)` means that you consdider the type-theoretic product of `X` and `Y`, let typeclass search give it an instance of a topological space, and finally apply `Top.of` to obtain the bundled object in `Top`.

If you make this change, then the lemmas below would probably need to be changed as well.